app/javascript/utils/EagerPagination.js

```diff
@@ -26,6 +28,10 @@ class EagerPagination {
     this.server.currentPage = -1
   }
 
+  isOverLimit (page) {
+    return (page * this.eager.size) >= (MAX_SERVER_LIMIT + SERVER_PAGE_SIZE)
```

Comment:
oh, I see here. I think it's easier to just have it be >= MAX_SERVER_LIMIT and set MAX_SERVER_LIMIT to 2100, then it'll be correct when you display the message too